### PR TITLE
[Merged by Bors] - chore: golf some proofs introduced in #6360

### PR DESCRIPTION
**Pull request by ADedecker**
Notably, getting rid of some injectivity restrictions makes some later proofs simpler.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/LinearIndependent.lean [hunk 1]
```diff
@@ -222,35 +222,13 @@ theorem LinearIndependent.map (hv : LinearIndependent R v) {f : M →ₗ[R] M'}
 
 /-- If `v` is an injective family of vectors such that `f ∘ v` is linearly independent, then `v`
     spans a submodule disjoint from the kernel of `f` -/
-theorem Submodule.ker_range_disjoint {f : M →ₗ[R] M'} (hi : v.Injective)
+theorem Submodule.range_ker_disjoint {f : M →ₗ[R] M'}
     (hv : LinearIndependent R (f ∘ v)) :
-    Disjoint (LinearMap.ker f) (Submodule.span R (Set.range v)) := by
```
Comment (ADedecker) [inline Mathlib/LinearAlgebra/LinearIndependent.lean:227]: I changed the order to match the lemma above, and changed the name accordingly

**Comment by ADedecker** [comment]
`family_injective_shortExact` is no longer used, is there any point in keeping it?

File: Mathlib/Algebra/Category/ModuleCat/Free.lean [hunk 2]
```diff
@@ -33,23 +33,23 @@ linear algebra, module, free
...
+  (hw : LinearIndependent R (g ∘ u ∘ Sum.inr))
   (hm : Mono f) (he : Exact f g) (huv : u ∘ Sum.inl = f ∘ v)
 
-theorem disjoint_span_sum : Disjoint (Submodule.span R (Set.range (u ∘ Sum.inl)))
-    (Submodule.span R (Set.range (u ∘ Sum.inr))) := by
-  rw [huv]
-  refine' Disjoint.mono_left (Submodule.span_mono (Set.range_comp_subset_range _ _)) _
-  rw [← LinearMap.range_coe, (Submodule.span_eq (LinearMap.range f)), (exact_iff _ _).mp he]
-  exact Submodule.ker_range_disjoint (Function.Injective.comp hu Sum.inr_injective) hw
+theorem disjoint_span_sum : Disjoint (span R (range (u ∘ Sum.inl)))
```
Comment (eric-wieser) [inline Mathlib/Algebra/Category/ModuleCat/Free.lean:47]: Can you move this to ` LinearAlgebra/Span`?

Comment (eric-wieser) [reply Mathlib/Algebra/Category/ModuleCat/Free.lean:47]: Ah nevermind, I think I might be missing something.

Comment (dagurtomas) [reply Mathlib/Algebra/Category/ModuleCat/Free.lean:47]: It uses exact sequences so it has to be stated about objects of `ModuleCat`, no?

**Comment by dagurtomas** [comment]
> `family_injective_shortExact` is no longer used, is there any point in keeping it?

I removed it

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/Algebra/Category/ModuleCat/Free.lean [hunk 3]
```diff
@@ -170,31 +145,27 @@ theorem span_rightExact {w : ι' → P} (hv : ⊤ ≤ Submodule.span R (Set.rang
 
 end Span
 
+noncomputable
+def Basis.ofShortExact {M : ModuleCat R} {f : N ⟶ M} {g : M ⟶ P} (h : ShortExact f g)
```
[the remark this answers is not in the document]
Comment (dagurtomas) [reply Mathlib/Algebra/Category/ModuleCat/Free.lean:151]: Added a docstring

**Comment by ADedecker** [comment]
I merged master to make sure that everything compiles with all of the recent technical changes.

**Comment by dagurtomas** [comment]
Looks good now, right? Can bors try again?

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**